### PR TITLE
Publish prep

### PR DESCRIPTION
**Pull request by mistermoe**
## What's Changed
* explicitly set minimum compatible java version to java 17
* add jitpack config file to configure runner to use java 17
* remove redundant version declarations in packages
* fix message and resource digest computations
    * sha-256 hashing the canonicalized json was missing
* `verify` - default `assertionMethods` to an empty list if null

### CODE REVIEW

**Comment by mistermoe** [comment]
@jiyoontbd thoughts on making e2e file an integration test of `http-client`? we can do so by adding an `integration-tests` package to `http-client/src`. Otherwise we'll need to shuffle some stuff around in the `build.gradle.kts` files so that the `e2e` package isn't published.

**Comment by mistermoe** [comment]
@michaelneale is the current state of `httpserver` taking `ktor.server` for a test drive to see what it's capable of? if so, thoughts on moving your findings to a branch or issue? thinking about removing the module until there's some meat on the bones

**Comment by michaelneale** [comment]
Yep the server is just a placeholder - can move it

**Comment by michaelneale** [comment]
@mistermoe yeah after you merge (or part of this) could move it out, or just don't publish the server? https://github.com/TBD54566975/tbdex-kt/issues/23 is the issue BTW

File: httpserver/src/test/kotlin/tbdex/sdk/httpserver/ServerTest.kt [hunk 1]
```diff
@@ -6,7 +6,6 @@ import io.ktor.server.application.Application
 import io.ktor.server.testing.handleRequest
 import io.ktor.server.testing.withTestApplication
 import org.junit.jupiter.api.Test
-import tbdex.server.tbdex.sdk.http_server.module
```
Comment (KendallWeihe) [inline httpserver/src/test/kotlin/tbdex/sdk/httpserver/ServerTest.kt:9]: Should we add `UnusedImports` to our detekt config? @phoebe-lew any thoughts?

```yml
style:
  UnusedImports:
    active: false
```

**Review by KendallWeihe — APPROVED** [review]

**Review by tomdaffurn — APPROVED** [review]

**Comment by jiyoontbd** [comment]
> @jiyoontbd thoughts on making e2e file an integration test of `http-client`? we can do so by adding an `integration-tests` package to `http-client/src`. Otherwise we'll need to shuffle some stuff around in the `build.gradle.kts` files so that the `e2e` package isn't published.

Totally good with me if you'd like to move the e2e.kt into httpclient in this PR :)

File: protocol/src/main/kotlin/tbdex/sdk/protocol/CryptoUtils.kt [hunk 2]
```diff
@@ -51,7 +51,7 @@ object CryptoUtils {
     // or just `#fragment`. See: https://www.w3.org/TR/did-core/#relative-did-urls.
     // Using a set for fast string comparison. DIDs can be long.
     val verificationMethodIds = setOf(parsedDidUrl.didUrlString, "#${parsedDidUrl.fragment}")
-    val assertionMethods = didResolutionResult.didDocument.assertionMethodVerificationMethodsDereferenced
+    val assertionMethods = didResolutionResult.didDocument.assertionMethodVerificationMethodsDereferenced ?: listOf()
```
Comment (jiyoontbd) [inline protocol/src/main/kotlin/tbdex/sdk/protocol/CryptoUtils.kt:54]: could also do `emptyList()` :)

Comment (mistermoe) [reply protocol/src/main/kotlin/tbdex/sdk/protocol/CryptoUtils.kt:54]: oo good call!

**Comment by KendallWeihe** [comment]
Do we want to go ahead and bump to 

```kotlin
implementation("com.github.TBD54566975:web5-kt:0.0.5-beta")
```

**Comment by mistermoe** [comment]
yep good call @KendallWeihe. on it

File: protocol/src/main/kotlin/tbdex/sdk/protocol/CryptoUtils.kt (hunk 2 above)
Comment (mistermoe) [reply protocol/src/main/kotlin/tbdex/sdk/protocol/CryptoUtils.kt]: done https://github.com/TBD54566975/tbdex-kt/pull/64/commits/802f4ce36cd5dc1e703560d064de7348dc9fb033 thanks @jiyoontbd !

**Comment by mistermoe** [comment]
@KendallWeihe bumped. thoughts on surfacing `web5-kt` via `api` dependency instead of `implementation`? prevents people from having to include both as deps

**Comment by KendallWeihe** [comment]
> @KendallWeihe bumped. thoughts on surfacing `web5-kt` via `api` dependency instead of `implementation`? prevents people from having to include both as deps

Yeah it's the same question we're puzzling over in `tbdex-js` with regards to the peer dependencies. I think it's sort of a six of one, half a dozen to the other. I do think I side with the peer dependency approach because of one particular reason, I outlined it in [this comment](https://github.com/TBD54566975/tbdex-js/pull/52#issuecomment-1777232367).

> I thought more about this and am fully bought in with this approach. The primary reason being Developer experience. Sure it's a slightly less good experience to have to install two packages. But, it makes the distinction between tbdex the transport agnostic protocol and the specific implementation over http much less ambiguous. Which is to say we're making it clear that tbdex isn't married to http.

**Pull request merged**